### PR TITLE
Minimal circles pathing

### PR DESCRIPTION
**Pull request by Gabrieldowen**
to run uncomment the test at the very bottom of the file

getPath inputs:
startLat = start airport latitude, 
startLong = start airport longitude, 
destLat = destination airport latitude, 
destLong = destination airport longitude, 
radius = radius of each API call, 
pathWidth = the width of the flight path that you want NOTAMs from

How it works:
1. takes start and destination coordinates
2. gets total distance between them
3. gets the distance the circles need to be spaced apart
4. gets the bearing or direction from the starting point to the destination
5. loops for the number of circles needed
6. for each loop get a circle 'stepDistance' away from the previous one
7. store in coordList
8. temporarily print the output

### CODE REVIEW

File: MinimalCirclesPath.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    # TODO For actual application you would need to capture N/S/E/W
+
+    # list of coordinates returned to call API with
+    coordList = [(startLat, startLong)]
+      
+    # gets total distance from start to finish
+    totalDistance = getDistance(startLat, startLong, destLat, destLong)
+
+    # gets step distance (pythagorean theorum)
+    stepDistance = sqrt((radius)**2-(pathWidth/2)**2)
```
Comment (n-grounds) [inline MinimalCirclesPath.py:76]: I'm confused why pathWidth is a factor in computer the step distance.  It looks like this is computing the long side of a right triangle where the other two sides are the radius and pathWidth/2 ... not sure I understand why

**Comment by n-grounds** [comment]
Great job figuring out the haversine algorithm for computing distance between points on a sphere.  The math looks right for that, but honestly it's hard to sight read all the sines/cosines/tangents and confirm it's all in the right order and everything :)

**Comment by n-grounds** [comment]
I'd love to see some unit tests for this code.  At the very least, let's vet that if we call getPath with two points "very close" to one another that we get back a two-element list and match the origin and destination with no points in between vs. calling getPath with two "further away from one another" points that require some intermediate steps.

File: app.py [hunk 2]
```diff
@@ -11,8 +12,13 @@ def index():
     if request.method == 'POST':
         NotamRequest = Models.NotamRequest(request.form)
 
-        # pass the form to getNOTAM to make API request
-        # apiOutput = GetNOTAM.getNOTAM(NotamRequest)
+        # get lat/long of airports
```
Comment (n-grounds) [inline app.py:15]: is there an issue entered in Github for figuring out how to determine this?

File: MinimalCirclesPath.py (hunk 1 above)
Comment (Gabrieldowen) [reply MinimalCirclesPath.py:76]: my logic is that there is a right triangle from the starting point, to the half way point between the two calls, and then the point where the path and both circles intersect. Im attaching a photo but I think I forgot to multiple the step distance by 2
tempImagea0Ur87

File: app.py (hunk 2 above)
Comment (Gabrieldowen) [reply app.py:15]: we wrote it on the board in the last meeting and I put it on the google doc notes for the meeting. But no not on github

File: MinimalCirclesPath.py (hunk 1 above)
Comment (n-grounds) [reply MinimalCirclesPath.py:76]: that makes sense, thanks

you'll find that details like this are worth documenting as inevitably someone down the road a few weeks (or months) will look at this code and have the same question yet again

Comment (Gabrieldowen) [reply MinimalCirclesPath.py:76]: Your welcome!

**Pull request merged**